web/src/js/ducks/options.ts

```diff
@@ -36,27 +36,27 @@ const reducer: Reducer<OptionsState> = (state = defaultState, action) => {
 }
 export default reducer
 
-export function pureSendUpdate(option: Option, value, dispatch) {
+export function pureSendUpdate(option: Option, value) {
     return async dispatch => {
         try {
             const response = await fetchApi.put('/options', {[option]: value});
             if (response.status === 200) {
                 dispatch(optionsEditorActions.updateSuccess(option))
             } else {
-                throw await response.text()
+                throw await response.text
             }
         } catch (error) {
             return dispatch(optionsEditorActions.updateError(option, error))
         }
     }
 }
 
-let sendUpdate = _.throttle(pureSendUpdate, 500, {leading: true, trailing: true})
-
 export function update(name: Option, value: any): AppThunk {
+    let sendUpdate = _.throttle(pureSendUpdate(name, value), 500, {leading: true, trailing: true})
```

Comment:
This looks wrong. If we reconstruct `_.throttle` on every invocation, it's not throttling.

Reply:
I see.
But we need option name and value to construct sendUpdate.
How can I get them?

Reply:
Maybe I'm not understanding the problem. What was wrong with the implementation before your changes?

Reply:
Sorry, I was doing wrong.
I fixed this.

Reply:
The following is the previous code.
It's passing dispatch as argument, but this is not corresponding to dispatch argument in the returned function.
So, dispatch function becomes undefined.
When we type in the input field, it doesn't execute dispatch and "intercept" state doesn't change.
Thus, we can't intercept any websites.

`export function pureSendUpdate(option: Option, value, dispatch) {
    return async dispatch => {
        try {
            const response = await fetchApi.put('/options', {[option]: value});
            if (response.status === 200) {
                dispatch(optionsEditorActions.updateSuccess(option))
            } else {
                throw await response.text()
            }
        } catch (error) {
            return dispatch(optionsEditorActions.updateError(option, error))
        }
    }
}`